### PR TITLE
Fix jinja2 dependency

### LINKED ISSUE #285: buildstock_eagle fails for dev env
**Describe the bug**
I log into Eagle, create a dev env for bsb develop, run a test ResStock yml. Simulations are successful, but then postprocessing (as far as I can tell) fails.

**To Reproduce**

1.
```
[jrobert1@el3 buildstockbatch]$ git branch
* develop
```

2.
```
[jrobert1@el3 buildstockbatch]$ bash create_eagle_env.sh -d dev-jrobert1-hpxml
Creating /shared-projects/buildstock/envs/dev-jrobert1-hpxml
...
```

3.
```
[jrobert1@el3 buildstockbatch]$ module load conda
[jrobert1@el3 buildstockbatch]$ source activate /shared-projects/buildstock/envs/dev-jrobert1-hpxml/
```

4.
```
(dev-jrobert1-hpxml) [jrobert1@el3 resstock]$ buildstock_eagle ~/resstock/project_national/national_upgrades2.yml
```

**Expected behavior**
No error. And to not have to do extra "pip install" hacks to get things to work.

**Logs**

```
####################
Traceback (most recent call last):
  File "/home/jrobert1/buildstockbatch/buildstockbatch/utils.py", line 98, in run_with_error_capture
    return func(*args, **kwargs)
  File "/home/jrobert1/buildstockbatch/buildstockbatch/eagle.py", line 740, in main
    batch.process_results()
  File "/home/jrobert1/buildstockbatch/buildstockbatch/base.py", line 603, in process_results
    postprocessing.combine_results(fs, self.results_dir, self.cfg, do_timeseries=do_timeseries)
  File "/home/jrobert1/buildstockbatch/buildstockbatch/postprocessing.py", line 570, in combine_results
    dask.compute(map(concat_partial, *zip(*enumerate(bldg_id_groups)), partition_vals_list))
  File "/shared-projects/buildstock/envs/dev-jrobert1-hpxml/lib/python3.7/site-packages/distributed/client.py", line 5403, in __exit__
    client.sync(self.__aexit__, type, value, traceback, code=code)
  File "/shared-projects/buildstock/envs/dev-jrobert1-hpxml/lib/python3.7/site-packages/distributed/utils.py", line 310, in sync
    self.loop, func, *args, callback_timeout=callback_timeout, **kwargs
  File "/shared-projects/buildstock/envs/dev-jrobert1-hpxml/lib/python3.7/site-packages/distributed/utils.py", line 376, in sync
    raise exc.with_traceback(tb)
  File "/shared-projects/buildstock/envs/dev-jrobert1-hpxml/lib/python3.7/site-packages/distributed/utils.py", line 349, in f
    result = yield future
  File "/shared-projects/buildstock/envs/dev-jrobert1-hpxml/lib/python3.7/site-packages/tornado/gen.py", line 762, in run
    value = future.result()
  File "/shared-projects/buildstock/envs/dev-jrobert1-hpxml/lib/python3.7/site-packages/distributed/client.py", line 5392, in __aexit__
    start=self.start, last_count=self.last_count, code=code, mode=self.mode
  File "/shared-projects/buildstock/envs/dev-jrobert1-hpxml/lib/python3.7/site-packages/distributed/core.py", line 905, in send_recv_from_rpc
    return await send_recv(comm=comm, op=key, **kwargs)
  File "/shared-projects/buildstock/envs/dev-jrobert1-hpxml/lib/python3.7/site-packages/distributed/core.py", line 698, in send_recv
    raise exc.with_traceback(tb)
  File "/shared-projects/buildstock/envs/dev-jrobert1-hpxml/lib/python3.7/site-packages/distributed/core.py", line 525, in handle_comm
    result = await result
  File "/shared-projects/buildstock/envs/dev-jrobert1-hpxml/lib/python3.7/site-packages/distributed/scheduler.py", line 7661, in performance_report
    profile_to_figure, (compute, scheduler, workers)
  File "/shared-projects/buildstock/envs/dev-jrobert1-hpxml/lib/python3.7/site-packages/distributed/scheduler.py", line 7657, in profile_to_figure
    figure, source = profile.plot_figure(data, sizing_mode="stretch_both")
  File "/shared-projects/buildstock/envs/dev-jrobert1-hpxml/lib/python3.7/site-packages/distributed/profile.py", line 384, in plot_figure
    from bokeh.plotting import ColumnDataSource, figure
  File "/home/jrobert1/.local/lib/python3.7/site-packages/bokeh/plotting/__init__.py", line 67, in <module>
    from ..document import Document; Document
  File "/home/jrobert1/.local/lib/python3.7/site-packages/bokeh/document/__init__.py", line 35, in <module>
    from .document import DEFAULT_TITLE ; DEFAULT_TITLE
  File "/home/jrobert1/.local/lib/python3.7/site-packages/bokeh/document/document.py", line 49, in <module>
    from ..core.templates import FILE
  File "/home/jrobert1/.local/lib/python3.7/site-packages/bokeh/core/templates.py", line 42, in <module>
    from jinja2 import Environment, FileSystemLoader, Markup
ImportError: cannot import name 'Markup' from 'jinja2' (/shared-projects/buildstock/envs/dev-jrobert1-hpxml/lib/python3.7/site-packages/jinja2/__init__.py)

In file: /home/jrobert1/buildstockbatch/buildstockbatch/utils.py, module run_with_error_capture line: 98
Local Variables: 'args':()
'kwargs':{}
'f':<_io.TextIOWrapper name='buildstockbatch_crash_details.log' mode='a' encoding='UTF-8'>
'text':'
####################
'
'func':<function main at 0x2ada82663b00>
'output_file':'buildstockbatch_crash_details.log'
```

**Platform (please complete the following information):**
- Simulation platform: **Eagle**
- BuildStockBatch version, branch, or sha: **develop**

```
(dev-jrobert1-hpxml) [jrobert1@el3 buildstockbatch]$ pip freeze
awsretry==1.0.1
bokeh==2.2.3
boto==2.49.0
boto3==1.17.11
botocore==1.20.11
buildstock-batch==0.15
-e git+https://github.com/NREL/buildstockbatch.git@b561790bdf9ec7c71a2ec184b2a0bfc57574317a#egg=buildstockbatch
certifi==2021.10.8
charset-normalizer==2.0.12
click==7.1.2
cloudpickle==1.6.0
cytoolz==0.11.2
dask @ file:///home/conda/feedstock_root/build_artifacts/dask-core_1644602974678/work
distributed @ file:///home/conda/feedstock_root/build_artifacts/distributed_1644612617784/work
docker==4.4.3
docutils==0.14
fsspec==0.8.5
HeapDict==1.0.1
idna==3.3
importlib-metadata @ file:///home/conda/feedstock_root/build_artifacts/importlib-metadata_1648728291958/work
Jinja2 @ file:///home/conda/feedstock_root/build_artifacts/jinja2_1648299710939/work
jmespath==0.10.0
joblib==1.0.1
locket==0.2.1
MarkupSafe @ file:///home/conda/feedstock_root/build_artifacts/markupsafe_1648737551960/work
msgpack==1.0.2
numpy @ file:///home/conda/feedstock_root/build_artifacts/numpy_1649806299270/work
packaging @ file:///home/conda/feedstock_root/build_artifacts/packaging_1637239678211/work
pandas==1.3.5
partd==1.1.0
Pillow @ file:///home/conda/feedstock_root/build_artifacts/pillow_1649817984594/work
psutil==5.8.0
pyarrow==7.0.0
pyparsing @ file:///home/conda/feedstock_root/build_artifacts/pyparsing_1649603503565/work
python-dateutil @ file:///home/conda/feedstock_root/build_artifacts/python-dateutil_1626286286081/work
pytz @ file:///home/conda/feedstock_root/build_artifacts/pytz_1647961439546/work
PyYAML==5.4.1
requests==2.27.1
ruamel.yaml==0.16.12
ruamel.yaml.clib==0.2.2
s3fs==0.4.2
s3transfer==0.3.4
six @ file:///home/conda/feedstock_root/build_artifacts/six_1620240208055/work
sortedcontainers==2.3.0
tblib==1.7.0
testfixtures==6.17.1
toolz==0.11.1
tornado @ file:///home/conda/feedstock_root/build_artifacts/tornado_1648827244717/work
typing-extensions==3.7.4.3
urllib3==1.26.9
websocket-client==0.57.0
yamale==3.0.4
zict==2.0.0
zipp @ file:///home/conda/feedstock_root/build_artifacts/zipp_1649012893348/work
```

### PR DESCRIPTION
**Pull request by joseph-robertson**
Fixes #285.

Fix `jinja2<3.1.0` per https://github.com/holoviz/panel/issues/3260.

## Pull Request Description

description here

## Checklist

Not all may apply

- [ ] Code changes (must work)
- [ ] Tests exercising your feature/bug fix (check coverage report on Checks -> BuildStockBatch Tests -> Artifacts)
- [ ] Coverage has increased or at least not decreased. Update `minimum_coverage` in `.github/workflows/ci.yml` as necessary.
- [ ] All other unit tests passing
- [ ] Update validation for project config yaml file changes
- [ ] Update existing documentation
- [ ] Run a small batch run to make sure it all works (local is fine, unless an Eagle specific feature)
- [ ] Add to the changelog_dev.rst file and propose migration text in the pull request

### CODE REVIEW

File: create_eagle_env.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ MY_CONDA_PREFIX="$CONDA_ENVS_DIR/$MY_CONDA_ENV_NAME"
 echo "Creating $MY_CONDA_PREFIX"
 module load conda
 conda remove -y --prefix "$MY_CONDA_PREFIX" --all
-conda create -y --prefix "$MY_CONDA_PREFIX" -c conda-forge "pyarrow>=3.0.0" python=3.7.8 "numpy>=1.20.0" "pandas>=1.0.0,!=1.0.4" "dask>=2021.5" "distributed>=2021.5" ruby
+conda create -y --prefix "$MY_CONDA_PREFIX" -c conda-forge "pyarrow>=7.0.0" python=3.7.8 "numpy>=1.20.0" "pandas>=1.0.0,!=1.0.4" "dask>=2021.5" "distributed>=2021.5" ruby
```
Comment (nmerket) [inline create_eagle_env.sh:29]:
```suggestion
conda create -y --prefix "$MY_CONDA_PREFIX" -c conda-forge "pyarrow>=7.0.0" python=3.7.8 "numpy>=1.20.0" "pandas>=1.0.0,!=1.0.4" "dask>=2021.5" "distributed>=2021.5" "jinja2<3.1.0" ruby
```

**Review by nmerket — COMMENTED** [review]
Did that work for you? I'd add it to the conda script as well so that there's not a conflict where conda installs a newer version on jinja and then pip installs an older version on top of it. That could lead to trouble.

File: create_eagle_env.sh (hunk 1 above)
Comment (joseph-robertson) [reply create_eagle_env.sh:29]: I tried this (without the change to `setup.py`), and it didn't work.

**Comment by joseph-robertson** [comment]
> Did that work for you? I'd add it to the conda script as well so that there's not a conflict where conda installs a newer version on jinja and then pip installs an older version on top of it. That could lead to trouble.

Yes, this worked for me.

**Review by nmerket — APPROVED** [review]

**Pull request merged**